### PR TITLE
Add pytest-pylint

### PR DESCRIPTION
**Pull request by holgern**
https://github.com/carsongee/pytest-pylint
https://pypi.python.org/pypi/pytest-pylint

### CODE REVIEW

File: recipes/pytest-pylint/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
+
+requirements:
+  build:
+    - python
+    - pip
```
Comment (rth) [inline recipes/pytest-pylint/meta.yaml]: Looking at `setup.py`, to run it only the two above packages are necessary. The rest can be removed from the build requirements.

File: recipes/pytest-pylint/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    - six
+    - pylint
+    - pytest
+    - astroid
+    - isort
+    - mccabe
+    - colorama
+    - py
+    - lazy-object-proxy
+    - wrapt
```
Comment (rth) [inline recipes/pytest-pylint/meta.yaml]: Why are there so much more dependencies than in [`setup.py`](https://github.com/carsongee/pytest-pylint/blob/master/setup.py#L22)

File: recipes/pytest-pylint/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+about:
+  home: https://github.com/carsongee/pytest-pylint
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: 'pytest plugin for running pylint against your codebase'
+
+  description: |
+    Run pylint with pytest and have configurable rule types (i.e. Convention, Warn, and Error) fail the build. You can also specify a pylintrc file.
```
Comment (rth) [inline recipes/pytest-pylint/meta.yaml]: Please keep the line below 80 characters. (Going to the newline in this block won't make an actual newline in the description.

File: recipes/pytest-pylint/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  imports:
+    - pytest_pylint
+
+about:
+  home: https://github.com/carsongee/pytest-pylint
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: 'pytest plugin for running pylint against your codebase'
+
```
Comment (rth) [inline recipes/pytest-pylint/meta.yaml]: newline can be removed.

File: recipes/pytest-pylint/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
```
Comment (rth) [inline recipes/pytest-pylint/meta.yaml]: It should be possible to use [`noarch`](https://conda-forge.org/docs/meta.html#building-noarch-packages) here, right?

Comment (holgern) [reply recipes/pytest-pylint/meta.yaml]: Yes, I try.

File: recipes/pytest-pylint/meta.yaml (hunk 4 above)
Comment (holgern) [reply recipes/pytest-pylint/meta.yaml]: Ok

File: recipes/pytest-pylint/meta.yaml (hunk 3 above)
Comment (holgern) [reply recipes/pytest-pylint/meta.yaml]: Done

File: recipes/pytest-pylint/meta.yaml (hunk 1 above)
Comment (holgern) [reply recipes/pytest-pylint/meta.yaml]: I will test.

Comment (holgern) [reply recipes/pytest-pylint/meta.yaml]: install_requires=['pytest>=2.7', 'pylint>=1.4.5', 'six'],
Are the requirements (setup.py line 22).

Comment (dougalsutherland) [reply recipes/pytest-pylint/meta.yaml]: You should put the version dependencies here too, because they probably matter. `pytest >=2.7`, `pylint >=1.4.5`.

Comment (holgern) [reply recipes/pytest-pylint/meta.yaml]: Thank, I changed it.

**Comment by dougalsutherland** [comment]
Okay, this looks good to me now. Thanks @rth for the review!

I'm not 100% sure how the new pip installation line works with entry points. I'm going to merge, but please check that it actually works to use with `py.test` once it's uploaded; if not, open an issue on the feedstock and @ me, and I'll help figure out how to fix it.

**Pull request merged**